tests/unit_tests/data_steward/tools/delete_stale_test_datasets_test.py

```diff
@@ -0,0 +1,131 @@
+"""
+Unit test for delete_stale_test_buckets module
+"""
+
+# Python imports
+import os
+from unittest import TestCase
+from unittest.mock import patch, Mock
+from datetime import datetime, timedelta, timezone
+
+# Third party imports
+
+# Project imports
+from tools import delete_stale_test_datasets
+
+
+class DeleteStaleTestDatasetsTest(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        self.mock_old_dataset_list_item_1 = Mock()
+        self.mock_old_dataset_list_item_1.dataset_id = 'all_of_us_dummy_old_dataset_1'
+
+        self.mock_old_dataset_list_item_2 = Mock()
+        self.mock_old_dataset_list_item_2.dataset_id = 'all_of_us_dummy_old_dataset_2'
+
+        self.mock_new_dataset_list_item = Mock()
+        self.mock_new_dataset_list_item.dataset_id = 'all_of_us_dummy_new_dataset'
+
+        self.dataset_list_items = [
+            self.mock_old_dataset_list_item_1,
+            self.mock_old_dataset_list_item_2, self.mock_new_dataset_list_item
+        ]
+
+        self.mock_old_dataset_1 = Mock()
+        self.mock_old_dataset_1.created = datetime.now(
+            timezone.utc) - timedelta(days=365)
+
+        self.mock_old_dataset_2 = Mock()
+        self.mock_old_dataset_2.created = datetime.now(
+            timezone.utc) - timedelta(days=180)
+
+        self.mock_new_dataset = Mock()
+        self.mock_new_dataset.created = datetime.now(
+            timezone.utc) - timedelta(days=1)
+
+        self.datasets = [
+            self.mock_old_dataset_1, self.mock_old_dataset_2,
+            self.mock_new_dataset
+        ]
+
+        self.mock_table = Mock()
+
+    @patch('tools.delete_stale_test_datasets.bq')
+    def test_check_project(self, mock_bq):
+        mock_bq.get_client.project = os.environ.get('GOOGLE_CLOUD_PROJECT')
```

Comment:
This env var will not be set in unit tests because it shouldn't need to connect to BQ. You can remove this test or use a different workaround. Ideally no env vars will be set. They are only to be used in integration tests.